cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -126,6 +132,87 @@ public CloseableIterator<StructuredRow> scan(Range keyRange, int limit) throws I
                              limit);
   }
 
+
+  /*
+   *  Sorting in memory for the no-sql implementation. We sort post table scan.
+   * */
+  @Override
+  public CloseableIterator<StructuredRow> scan(Range keyRange, int limit, String orderByField, SortOrder sortOrder)
+    throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: Scan range {} with limit {}, sort field {} and sort order {}", schema.getTableId(),
+              keyRange, limit, orderByField, sortOrder);
+    if (!schema.isIndexColumn(orderByField)) {
+      throw new InvalidFieldException(schema.getTableId(), orderByField, "is not an indexed column");
+    }
+    Comparator<StructuredRow> comparator = sortOrder.equals(SortOrder.ASC) ? getComparator(orderByField) :
+      getComparator(orderByField).reversed();
+    PriorityQueue<StructuredRow> rows = new PriorityQueue<>(comparator);
+
+    try (ScannerIterator scannerIterator = new ScannerIterator(getScanner(keyRange), schema)) {
+      while (scannerIterator.hasNext()) {
+        rows.offer(scannerIterator.next());
+      }
+    }
+    // return an iterator for the sorted elements
+    AbstractCloseableIterator<StructuredRow> abstractCloseableIterator = new
+      AbstractCloseableIterator<StructuredRow>() {
+      @Override
```

Comment:
indentation

Reply:
this one too seems right?

Reply:
it should be 2 space indentation from the previous line, and your IDE should do it automatically if properly setup.

In this particular case, can just modify the earlier lines to be shorter. It's a bit weird to have a line end with `new `

```
CloseableIterator<StructuredRow> iter = new AbstractCloseableIterator<StructuredRow>() {

}
```
In general, when you're declaring a variable it should be the most generic interface/class possible. For example, Map instead of HashMap, CloseableIterator instead of AbstractCloseableIterator.